src/hydra/dataProvider.ts

```diff
@@ -604,6 +609,51 @@ function dataProvider(
         ),
       );
 
+  /*
+   * The fetchAllPages method allows running as many requests as needed in order to load all pages of a list.
+   * This function uses the already transformed react-admin response data and could be improved by using the hydra pagination
+   */
+  const fetchAllPages = async (
+    resource: string,
+    params: Partial<GetListParams>,
+    previousResult?: GetListResult,
+  ): Promise<GetManyResult> => {
+    params.pagination = {
+      // Using a default perPage value big enough to limit the number of requests
+      // If the API has a max_per_page value, the perPage parameter can be ignored
+      perPage: params.pagination?.perPage || 100,
+      page: params.pagination?.page || 1,
+    };
+
+    // Get the current page data
+    const pageResult = (await fetchApi(GET_LIST, resource, {
```

Comment:
`GET_LIST` could be a parameter (it could be used with `GET_MANY_REFERENCE` for instance).

Reply:
No, we really want to run the GET_LIST operation here, as it's the only one returning a data AND a total number of items.

Reply:
It seems to me that is also the case with `GetManyReferenceResult`.

Reply:
Ok, I can add allow both GET_LIST and GET_MANY_REFERENCE

Reply:
`export const fetchActionsWithTotalResponse = ['getList', 'getManyReference'];` => perfect :-)

Reply:
Done, can you check if it's okayfor you ?